### PR TITLE
fix(ovelay): fix error in terminal (Issue#1651)

### PR DESCRIPTION
**Pull request by denys-kolomiitsev**
**Description:**

- Fix error in terminal console "window not defined".

Issues:

- Issue #1651  

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

File: libs/overlay/src/lib/ChatOverlay.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+'use client';
```
Comment (mikitabut) [inline libs/overlay/src/lib/ChatOverlay.ts]: Is this really working? This is react specific thing or I'm wrong?

File: libs/overlay/src/lib/ChatOverlay.ts [hunk 2]
```diff
@@ -358,7 +360,7 @@ export class ChatOverlay {
    */
   destroy() {
     window.removeEventListener('message', this.process);
-    this.iframeInteraction.fail('Chat Overlay destroyed');
+    this.iframeInteraction.complete();
```
Comment (mikitabut) [inline libs/overlay/src/lib/ChatOverlay.ts]: Not sure that this is good idea. Actually, your iframe is not complete at this moment, but failing. Maybe we should just not log this as error, but still fail promise?
Moreover if this iframe interaction completed that means that user can send chat overlay options but this is not right scenario - they should do this only when it's really complete

File: libs/overlay/src/lib/ChatOverlay.ts (hunk 1 above)
Comment (IlyaBondar) [reply libs/overlay/src/lib/ChatOverlay.ts]: @mikitabut , it's NextJS feature

File: apps/chat/src/utils/app/overlay.ts [hunk 3]
```diff
@@ -21,6 +21,7 @@ export function sendPMResponse(
   type: OverlayRequests,
   requestParams: PostMessageRequestParams,
 ) {
+  if (typeof window === 'undefined') return;
```
Comment (IlyaBondar) [inline apps/chat/src/utils/app/overlay.ts:24]: can we move all `window`-related code in separate file and mark by `use client`?

File: libs/overlay/src/lib/ChatOverlay.ts (hunk 1 above)
Comment (mikitabut) [reply libs/overlay/src/lib/ChatOverlay.ts]: This ChatOverlay.ts can be loaded outside of NextJS, so we shouldn't use it here

Comment (mikitabut) [reply libs/overlay/src/lib/ChatOverlay.ts]: @IlyaBondar looks like its react :)
https://react.dev/reference/rsc/use-client

Comment (denys-kolomiitsev) [reply libs/overlay/src/lib/ChatOverlay.ts]: [This is Next JS feature](https://nextjs.org/docs/app/building-your-application/rendering/client-components#using-client-components-in-nextjs)

File: apps/chat/src/utils/app/overlay.ts (hunk 3 above)
Comment (denys-kolomiitsev) [reply apps/chat/src/utils/app/overlay.ts:24]: You a right, it's NextJS feature.
However, it doesn't work in this case.
For now only this solution works.

File: libs/overlay/src/lib/ChatOverlay.ts (hunk 2 above)
Comment (denys-kolomiitsev) [reply libs/overlay/src/lib/ChatOverlay.ts]: You are right. Reverted.

**Review by IlyaBondar — DISMISSED** [review]

**Review by mikitabut — APPROVED** [review]

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1644.nightly-test.deltixhub.io
>E2E tests status: success

**Pull request merged**